src/components/PostDetails/PostDetails.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useEffect } from 'react';
+import { NewCommentForm } from '../NewCommentForm';
+import './PostDetails.scss';
+import { Post, Comment } from '../../types';
+import { getComments, delComment } from '../../api/comments';
+import { getPostbyId } from '../../api/posts';
+
+type Props = {
+  postId: number | undefined;
+};
+
+export const PostDetails: React.FC<Props> = ({
+                                               postId,
+                                             }) => {
+  const [commentsList, setCommentsList] = useState<Comment[] | undefined>([]);
```

Comment:
The state could initialize simply as an empty array
```suggestion
  const [commentsList, setCommentsList] = useState<Comment[]>([]);
```